test/k23si/3SITxnTest.cpp

```diff
@@ -1794,10 +1797,690 @@ seastar::future<> testScenario02() {
+                        K2INFO("Value of k4: " << val1);
+                        K2INFO("Value of k5: " << val2);
+                        K2INFO("Value of k6: " << val3);
+                        K2EXPECT(status1, dto::K23SIStatus::KeyNotFound);
+                        K2EXPECT(status2, dto::K23SIStatus::OK);
+                        K2EXPECT(status2, dto::K23SIStatus::OK);
+                        K2EXPECT(val1.f1, "");
+                        K2EXPECT(val1.f2, "");
+                        K2EXPECT(val2, v0);
+                        K2EXPECT(val3, v0);
+                    });
+                });
+            });
+        }); // end do-with
+    }) // end case 07-10
+    .then([&] {
+        K2INFO("------- SC06.case11 ( After partial Finalization_abort, txn continues and then End_Abort all records ) -------");
+        return getTimeNow();
+    })
+    .then([this](dto::Timestamp&& ts) {
+        return seastar::do_with(
+            dto::K23SI_MTR {.txnid = txnids++, .timestamp = std::move(ts), .priority = dto::TxnPriority::Medium},
+            dto::Key {.schemaName = "schema", .partitionKey = "SC06_pkek4", .rangeKey = "rKey4"},
+            dto::Key {.schemaName = "schema", .partitionKey = "SC06_pkey5", .rangeKey = "rKey5"},
+            dto::Key {.schemaName = "schema", .partitionKey = "SC06_pkey6", .rangeKey = "rKey6"},
+            DataRec {.f1="SC05_f1_zero", .f2="SC04_f2_zero"},
+            [this](auto& mtr, auto& k4, auto& k5, auto& k6, auto& v0) {
+            return seastar::when_all(doWrite(k4, v0, mtr, k4, collname, false, true, ErrorCaseOpt::NoInjection), \
+                    doWrite(k5, v0, mtr, k5, collname, false, false, ErrorCaseOpt::NoInjection))
+            .then([](auto&& response) mutable {
+                auto& [resp1, resp2] = response;
+                auto [status1, val1] = resp1.get0();
+                auto [status2, val2] = resp2.get0();
+                K2INFO("SC06.case11::OP_WI_k4. " << "status: " << status1.code << " with MESG: " << status1.message);
+                K2INFO("SC06.case11::OP_WI_k5. " << "status: " << status2.code << " with MESG: " << status2.message);
+                K2EXPECT(status1, dto::K23SIStatus::Created);
+                K2EXPECT(status2, dto::K23SIStatus::Created);
+            })
+            .then([&] {
+                return doFinalize(k4, k4, mtr, collname, false, ErrorCaseOpt::NoInjection)
+                .then([](auto&& response)  {
+                    auto& [status, val] = response;
+                    K2INFO("SC06.case11::OP_finalize_abort_k4. " << "status: " << status.code << " with MESG: " << status.message);
+                    K2EXPECT(status, dto::K23SIStatus::OK);
+                });
+            })
+            .then([&] {
+                return doWrite(k6, v0, mtr, k4, collname, false, false, ErrorCaseOpt::NoInjection)
```

Comment:
Let's remove this write, then the test is a more realistic scenario where the abort-finalize is sent more than once as part of retry logic.